include vars for dispatch option

djdanielsson opened the pull request:
<!-- markdownlint-disable -->
# What does this PR do?
<!--- Brief explanation of the code or documentation change you've made -->
looking to support an interesting way to include vars from different sources and organize it all with dispatch

# How should this be tested?
<!--- Automated tests are preferred, but not always doable - especially for infrastructure. Include commands to run your new feature, and also post-run commands to validate that it worked. (please use code blocks to format code samples) -->
manually

# Is there a relevant Issue open for this?
<!--- Provide a link to any open issues that describe the problem you are solving. -->
no

# Other Relevant info, PRs, etc
<!--- Please provide link to other PRs that may be related (blocking, resolves, etc. etc.) -->
none

djdanielsson committed:
looking to support an interesting way to include vars from different sources and organize it all with dispatch

Tompage1994 commented:
I'm not sure I entirely understand what's going on here and what the use case is

Tompage1994 commented on roles/dispatch/defaults/main.yml:
```diff
@@ -1,4 +1,5 @@
 ---
+dispatch_sort_included_vars: false
```
I don't like the name of the variable as I don't think it really covers what it does.

We may want to collect ideas for this but I was thinking something more along the lines of `dispatch_include_wildcard_vars` or similar.

In particular we aren't really 'sorting' the vars in any way and really they don't need to be explicitly 'included' as they could be applied using inventories or any other mechanism really.

Tompage1994 commented on roles/dispatch/README.md:
```diff
@@ -8,6 +8,27 @@ An Ansible Role to run all roles in the infra.aap_configuration collection.
 
 This is a meta role, its purpose is to run the other roles in the collection, it does not run all of them, and can be used to call roles in a custom order, The control variable is the `aap_configuration_dispatcher_roles` which will pull roles from the different services and run them, If you wish to just run a subset of services from our default lists remove entries from this var.
 
+There is also an option `dispatch_sort_included_vars` which is default set to `false` but if true it expects you to have used include_vars dir option and the vars themselves to match the naming but add on to them for example `controller_templates_dev`.
```
Can we improve this readme a little more to make it clearer exactly what it does please? Perhaps the example could be more of a walkthrough of what actually happens. (i.e. you include/apply some vars -> they get concatenated to the main var -> they're applied to the role)

djdanielsson replied:
I tried explaining it a lot more, let me know what you think now

Tompage1994 commented on roles/dispatch/tasks/main.yml:
```diff
@@ -1,4 +1,257 @@
 ---
+- name: Sort out vars that are included
+  when: dispatch_sort_included_vars
+  block:
```
Can these all be separated out into a separate task file which is only included when the variable is set to true?

Tompage1994 replied:
e.g. `tasks/include_wildcard_vars.yml` (or matching whatever the var name ends up being)

Tompage1994 reviewed (changes requested):
A deeper look into it and Sean's explanation has helped. I get what's going on here and I think this is a really useful addition.

BUT I think we need better documentation and examples in that readme AND I think the name of the variable isn't great as it doesn't really capture what it's doing

Tompage1994 committed:
Merge branch 'devel' into dispatch_vars

djdanielsson committed:
renamed:    changelogs/fragments/async_fixes -> changelogs/fragments/async_fixes.yml
	new file:   changelogs/fragments/include_wildcard_vars.yml
	modified:   roles/dispatch/README.md
	modified:   roles/dispatch/defaults/main.yml
	new file:   roles/dispatch/tasks/include_wildcard_vars.yml
	modified:   roles/dispatch/tasks/main.yml

djdanielsson committed:
Merge branch 'dispatch_vars' of github.com:djdanielsson/infra.aap_configuration into dispatch_vars

djdanielsson committed:
Merge branch 'devel' into dispatch_vars

djdanielsson committed:
modified:   roles/dispatch/README.md

djdanielsson committed:
Merge branch 'dispatch_vars' of github.com:djdanielsson/infra.aap_configuration into dispatch_vars

djdanielsson committed:
modified:   roles/dispatch/tasks/include_wildcard_vars.yml

djdanielsson committed:
modified:   roles/dispatch/tasks/main.yml

djdanielsson committed:
modified:   changelogs/fragments/async_fixes.yml

djdanielsson committed:
modified:   roles/dispatch/README.md

djdanielsson committed:
modified:   roles/dispatch/README.md

djdanielsson committed:
modified:   roles/dispatch/README.md

djdanielsson committed:
modified:   roles/dispatch/README.md

djdanielsson committed:
modified:   roles/dispatch/README.md

djdanielsson committed:
modified:   roles/dispatch/tasks/include_wildcard_vars.yml

djdanielsson committed:
Merge branch 'devel' into dispatch_vars

djdanielsson committed:
new file:   changelogs/fragments/hub_publish.yml

Tompage1994 reviewed (approved):
LGTM
